File: daily_challenges/165.js
/**
 * @param {string} version1
 * @param {string} version2
 * @return {number}
 */
var compareVersion = function (version1, version2) {
    const getTokens = (version) => {
        return version.split(".")
    }
    const v1 = getTokens(version1)
    const v2 = getTokens(version2)
    const m = v1.length, n = v2.length
    let i = 0
    while (i < m || i < n) {
        const a = i < m ? parseInt(v1[i], 10) : 0
        const b = i < n ? parseInt(v2[i], 10) : 0
        if (a < b) {
            return -1
        } else if (a > b) {
            return 1
        }
        i += 1

    }
    return 0
};

File: daily_challenges/165.py
# Microsoft, Apple

# Tc = O(m+n)
# Sc = O(m+n)

class Solution:
    def getTokens(self, version: str) -> list[str]:
        return version.split('.')

    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = self.getTokens(version1)
        v2 = self.getTokens(version2)
        m, n = len(v1), len(v2)
        i = 0
        while i < m or i < n:
            a = int(v1[i]) if i < m else 0
            b = int(v2[i]) if i < n else 0
            if a < b:
                return -1
            elif a > b:
                return 1
            else:
                i += 1
        return 0
    

            
        
    
